pkg/ccl/importccl/read_import_mysql.go

```diff
@@ -139,5 +170,215 @@ func mysqlToCockroach(
 	default:
 		return nil, errors.Errorf("unexpected value type %T: %v", v, v)
 	}
+}
+
+func readMysqlCreateTable(
+	input io.Reader, evalCtx *tree.EvalContext, match string,
+) ([]*sqlbase.TableDescriptor, error) {
+	r := bufio.NewReaderSize(input, 1024*64)
+	tokens := mysql.NewTokenizer(r)
+
+	var ret []*sqlbase.TableDescriptor
+	var found []string
+	for {
+		stmt, err := mysql.ParseNext(tokens)
+		if err == io.EOF {
+			if match != "" {
+				return nil, errors.Errorf("table %q not found in file (found tables: %s)", match, strings.Join(found, ", "))
+			}
+			if ret == nil {
+				return nil, errors.Errorf("no table definition found")
+			}
+			return ret, nil
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "mysql parse error")
+		}
+		if i, ok := stmt.(*mysql.DDL); ok && i.Action == mysql.CreateStr {
+			name := i.NewName.Name.String()
+			if match != "" && match != name {
+				continue
+			}
+			id := sqlbase.ID(int(defaultCSVTableID) + len(ret))
+			tbl, err := mysqlTableToCockroach(evalCtx, id, name, i.TableSpec)
+			if err != nil {
+				return nil, err
+			}
+			if match == name {
+				return []*sqlbase.TableDescriptor{tbl}, nil
+			}
+			ret = append(ret, tbl)
+		}
+	}
+}
+
+func mysqlTableToCockroach(
+	evalCtx *tree.EvalContext, id sqlbase.ID, name string, in *mysql.TableSpec,
+) (*sqlbase.TableDescriptor, error) {
+	if in == nil {
+		return nil, errors.Errorf("could not read definition for table %q (possible unsupoprted type?)", name)
+	}
+
+	time := hlc.Timestamp{WallTime: evalCtx.GetStmtTimestamp().UnixNano()}
+	priv := sqlbase.NewDefaultPrivilegeDescriptor()
+	desc := sql.InitTableDescriptor(id, defaultCSVParentID, name, time, priv)
```

Comment:
we already have a parentID up in the import function. i think we should plumb it down to here so that we can change it in one place.

Reply:
Done.